### ISSUE TITLE
NullPointerException on Quarkus' NotFoundExceptionMapper

### ISSUE THREAD

**Issue by brunoabdon** [quarkusio/quarkus#3405]
**Describe the bug**
`NullPointerException` on `quarku's NotFoundExceptionMapper`

**Expected behavior**
Throwing `javax.ws.rs.NotFoundException` from methods in resource classes should not end up with NullPointerEception.

**Actual behavior**
Throwing `javax.ws.rs.NotFoundException` from methods in a resource classes end up with `NullPointerEception` on `quarku's NotFoundExceptionMapper` if:
- The method is defined in a superclass of the resource class; and
- The superclass has no `@Path` annotation

**To Reproduce**
1. Create class with a method `void failNow(){ throw new javax.ws.rs.NotFoundException()}
2. Create a resource class extending the above class annoted with `@Path("somepath")`.
3. Add a method annotated with `@GET` and calling `super.failNow()
4. Run de application
5. Access http://localhost:8080/somepath
6. See the NPE in the log

**Configuration**
N/A
**Screenshots**
N/A

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
Linux vasti 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: 
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode
 - GraalVM version (if different from Java): 
N/A
 - Quarkus version or git rev: 
0.20
**Additional context**

**Comment by gsmet** [comment]
Hi @brunoabdon ,

Any chance you could share the full stacktrace?

**Comment by brunoabdon** [comment]
> Hi @brunoabdon ,
> 
> Any chance you could share the full stacktrace?

Here it goes
<pre>
ERROR [io.und.req.io] (executor-thread-1) Exception handling request 01104086-d125-4925-95f0-6d863210962c-3 to /planinha/contas/58: java.lang.NullPointerException
	at io.quarkus.resteasy.runtime.NotFoundExceptionMapper$ResourceDescription.fromBoundResourceInvokers(NotFoundExceptionMapper.java:92)
	at io.quarkus.resteasy.runtime.NotFoundExceptionMapper.toResponse(NotFoundExceptionMapper.java:142)
	at io.quarkus.resteasy.runtime.NotFoundExceptionMapper.toResponse(NotFoundExceptionMapper.java:37)
	at org.jboss.resteasy.core.ExceptionHandler.executeExceptionMapper(ExceptionHandler.java:137)
	at org.jboss.resteasy.core.ExceptionHandler.unwrapException(ExceptionHandler.java:183)
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:104)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:496)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:249)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:60)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$8$1$1.call(UndertowDeploymentRecorder.java:489)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:197)
	at io.undertow.server.handlers.HttpContinueReadHandler.handleRequest(HttpContinueReadHandler.java:65)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:91)
	at io.undertow.server.handlers.CanonicalPathHandler.handleRequest(CanonicalPathHandler.java:49)
	at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup.handleHotDeploymentRequest(UndertowHotReplacementSetup.java:77)
	at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup$1$1.handleRequest(UndertowHotReplacementSetup.java:56)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:364)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:224)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
</pre>

**Comment by loicmathieu** [comment]
I can take care of this one as I updated this class parts on the ongoing PR for a JSON representation in this exception mapper: https://github.com/quarkusio/quarkus/pull/3401

**Comment by gsmet** [comment]
Please make it a separate PR. We need this one fixed independently of the
discussion on the other PR.

**Comment by loicmathieu** [comment]
Hi @brunoabdon can you send a reproducer or at least the two classes in question because I can see what can be wrong in the code but I cannot reproduce the error with the steps you describe in the issue

**Comment by brunoabdon** [comment]
Hi, @loicmathieu . The two classes are on this gist: 

https://gist.github.com/brunoabdon/bf4adf7d8f91a008e68300472c70434d

**Comment by loicmathieu** [comment]
@brunoabdon I tested the two classes on you gist with a fresh project with Quarkus 0.20 and there is no NPE. So I cannot reproduce your error.

I did the following:
1. Generate a fresh project with
```
mvn io.quarkus:quarkus-maven-plugin:0.20.0:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=rest-json \
    -DclassName="org.acme.rest.json.FruitResource" \
    -Dpath="/fruits" \
    -Dextensions="resteasy-jsonb"
```

2. Add your two classes in the `org.acme.rest.json` package and remove the generated `FruitResource` 

3. Launch it and test it ... and it works

So I will need a full reproducer to works on it, a zip of a project if possible.

**Comment by brunoabdon** [comment]
@loicmathieu, Indeed, the scenario is a little bit more elaborated than what I've thought.

The 3 classes on this gist reproduce this scenario using the _quarkus-maven-plugin:0.20.0:create_ steps you listed above.

https://gist.github.com/brunoabdon/ba5cbad0f815a8b26c074edde99eb629

The scenario we actually need is:

1. A `@Path` annotated class C1 throwing the `NotFoundException` from a `@GET` method; and 
2. An unrelated class C2 without `@Path` but with some `@GET`method; and
3. A `@Path` annotated class C3 extends C2.

While dealing with C1's exception, `NotFoundExceptionMapper` builds an html page listing all available endpoints. And there is where it wrongly expects C2 to have a `@Path` annotation, just because it has a `@GET` annotated method.

**Comment by brunoabdon** [comment]
TLDR; You want to replace [line 92 of `NotFoundExceptionMapper.java`](https://github.com/quarkusio/quarkus/blob/2a3d9dd98879ed3e01129042930620e5594882b1/extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java#L92) for:

`String basePath = (((ResourceMethodInvoker)entry.getValue().get(0)).getResourceClass()).getAnnotation(Path.class).value();             `

**Comment by loicmathieu** [comment]
@brunoabdon thanks for the provided gist, it triggers the NPE. I will look at it.
